### PR TITLE
#185629831 Add: Periodically fetch app config

### PR DESCRIPTION
**Pull request by keyur9**
Add: Periodically fetch app config
Bump version to 1.9.1

### CODE REVIEW

**Review by dgilling — APPROVED** [review]

File: moesifwsgi/workers.py [hunk 1]
```diff
@@ -173,3 +175,45 @@ def stop(self):
...
+    def __init__(self, debug, config):
+        self.DEBUG = debug
+        self.scheduler = None
+        self.config = config
+
+    def exit_config_job(self):
+        try:
+            # Shut down the scheduler
+            self.scheduler.remove_job('moesif_config_job')
+            self.scheduler.shutdown()
```
Comment (praves77) [inline moesifwsgi/workers.py:191]: add `self.scheduler = None` too?

File: moesifwsgi/workers.py [hunk 2]
```diff
@@ -173,3 +175,45 @@ def stop(self):
...
+        except Exception as ex:
+            if self.DEBUG:
+                print("Error during shut down of the config scheduler")
+                print(str(ex))
+
+    def schedule_background_job(self):
+        try:
+            if not self.scheduler:
+                self.scheduler = BackgroundScheduler(daemon=True)
+            if not self.scheduler.get_jobs():
```
Comment (praves77) [inline moesifwsgi/workers.py:201]: @keyur9 do we need to check if get_jobs() is `truthy` but job status is cancelled or terminated? In that case we need to clean up and reschedule?

Comment (praves77) [reply moesifwsgi/workers.py:201]: we can take care of it in next release.

Comment (keyur9) [reply moesifwsgi/workers.py:201]: get_jobs return the list of pending and scheduled jobs, so we're checking if no job is pending or scheduled then only we add the config job. So there won't be any need to check the status

File: moesifwsgi/workers.py (hunk 1 above)
Comment (keyur9) [reply moesifwsgi/workers.py:191]: It'd be fine since this would be trigger on job exit. So no need to set it to None

**Review by praves77 — APPROVED** [review]

File: moesifwsgi/workers.py [hunk 3]
```diff
@@ -173,3 +175,45 @@ def stop(self):
...
+                self.scheduler.start()
+                self.scheduler.add_job(
+                    func=lambda: self.config.update_configuration(),
+                    trigger=IntervalTrigger(seconds=60),
```
Comment (sam-h-bean) [inline moesifwsgi/workers.py:205]: so now this will get the app config every 60 seconds? great!

**Pull request merged**